source/grid/grid_tools.cc

```diff
@@ -5088,12 +5088,18 @@ namespace GridTools
     MPI_Comm                                  mpi_communicator)
   {
 #ifndef DEAL_II_WITH_MPI
-    (void)local_description;
-    (void)mpi_communicator;
-    Assert(false,
-           ExcMessage(
-             "GridTools::build_global_description_tree() requires MPI."));
-    return RTree<std::pair<BoundingBox<spacedim>, unsigned int>>{};
+    // Building a tree with the only boxes available without MPI
+    std::vector<std::pair<BoundingBox<spacedim>, unsigned int>> boxes_index(
+      local_description.size());
+    for (const auto &box : local_description)
+      {
+        // Adding to each box the rank of the process owning it
+        for (unsigned int i = 0; i < local_description.size(); ++i)
+          boxes_index[i] = std::make_pair(local_description[i], 0);
```

Comment:
```suggestion
          boxes_index[i] = std::make_pair(local_description[i], 0u);
```
This makes sure that the data type of the zero is correctly deduced and consequently that no conversion from the pair type on the rhs to the pair type on the lhs is necessary.